tee-worker/omni-executor/aa-contracts/src/accounts/SmartAccountFactory.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import "@openzeppelin/contracts/utils/Create2.sol";
+import "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+
+import "../interfaces/ISenderCreator.sol";
+import "./SmartAccount.sol";
+
+/**
+ * A sample factory contract for Account
+ * A UserOperations "initCode" holds the address of the factory, and a method call (to createAccount, in this sample factory).
+ * The factory's createAccount returns the target account address even if it is already installed.
+ * This way, the entryPoint.getSenderAddress() can be called either before or after the account is created.
+ */
+contract SmartAccountFactory {
+    SmartAccount public immutable accountImplementation;
+    ISenderCreator public immutable senderCreator;
+
+    constructor(IEntryPoint _entryPoint) {
+        accountImplementation = new SmartAccount(_entryPoint);
+        senderCreator = _entryPoint.senderCreator();
+    }
+
+    /**
+     * create an account, and return its address.
+     * returns the address even if the account is already deployed.
+     * Note that during UserOperation execution, this method is called only if the account is not deployed.
+     * This method returns an existing account address so that entryPoint.getSenderAddress() would work even after account creation
+     */
+    function createAccount(bytes32 oa, bytes32 clientId, address root) public returns (SmartAccount ret) {
```

Comment:
As mentioned, shall we store `clientId` into the entrypoint storage? Maybe even as ctor param